pkg/cli/node.go

```diff
@@ -0,0 +1,115 @@
+package cli
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/QMSTR/qmstr/pkg/builder"
+	"github.com/QMSTR/qmstr/pkg/common"
+	"github.com/QMSTR/qmstr/pkg/gnubuilder"
+	"github.com/QMSTR/qmstr/pkg/qmstr/service"
+	"github.com/spf13/cobra"
+	"golang.org/x/net/context"
+)
+
+var (
+	target, input []string
+	mode          gnubuilder.Mode
+)
+
+var nodeCmd = &cobra.Command{
+	Use:   "node",
+	Short: "Modify nodes in the database",
+	Long: `Node is used when we want to modify file nodes in the database.
+A child command has to be specified after the 'node' command, to define 
+the action to be taken in the database. For instance, to add or dump a node.`,
+	Run: func(cmd *cobra.Command, args []string) {},
+}
+
+var nodeAddCmd = &cobra.Command{
+	Use:   "add",
+	Short: "Add file nodes into the database",
+	Long:  `Add file nodes and the connection between them into the database.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		setUpBuildServer()
+		if err := addFileNodes(args); err != nil {
+			Log.Fatalf("Add file nodes failed: %v", err)
+		}
+		tearDownServer()
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(nodeCmd)
+	nodeCmd.AddCommand(nodeAddCmd)
+	nodeAddCmd.Flags().StringSliceVarP(&target, "output", "o", []string{}, "Output file name")
```

Comment:
Instead of using the cmd.Flags().StringSliceVarP() that requires a variable to exist, as in this case "target", is better to use cmd.Flags().StringSliceP("myFlag", ...) where you just name the flag and later you can retrieve it by the means of cmd.Flags().GetStringSlice("myFlag")